src/Utilities/ErrorHandling/FormatStacktrace.cpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Utilities/ErrorHandling/FormatStacktrace.hpp"
+
+#include <array>
+#include <boost/algorithm/string.hpp>  // For boost::to_lower
+#include <boost/core/demangle.hpp>
+#include <cstddef>
+#include <cstdlib>     // For std::getenv
+#include <execinfo.h>  // For backtrace_symbols
+// link.h is not available on all platforms
+#if __has_include(<link.h>)
+#include <link.h>
+#endif
+#include <memory>
+#include <string>
+
+namespace stacktrace_detail {
+
+namespace {
+bool env_var_to_bool(std::string env_var) {
+  boost::to_lower(env_var);
+  return not(env_var == "false" or env_var == "no");
+}
+}  // namespace
+
+std::string abbreviated_symbol_name(const std::string& symbol_name) {
+  constexpr size_t abbrev_length_begin = 300;
+  constexpr size_t abbrev_length_end = 100;
```

Comment:
could you and `_from_end` and `_from_start`? I found the names pretty unclear until I read how they were used.